instaclustr/resource_cluster.go

```diff
@@ -325,29 +365,165 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 		createData.RackAllocation = &rackAllocation
 	}
 
-	var jsonStr []byte
-	jsonStr, err = json.Marshal(createData)
+	var jsonStrCreate []byte
+	jsonStrCreate, err = json.Marshal(createData)
 	if err != nil {
 		return formatCreateErrMsg(err)
 	}
 
-	id, err := client.CreateCluster(jsonStr)
+	id, err := client.CreateCluster(jsonStrCreate)
 	if err != nil {
 		return formatCreateErrMsg(err)
 	}
 	d.SetId(id)
 	d.Set("cluster_id", id)
 	log.Printf("[INFO] Cluster %s has been created.", id)
-	return nil
+
```

Comment:
This method is getting rather large, could it please be split out into smaller well named methods?